src/app/owner/pets/dashboard/page.test.tsx

```diff
@@ -0,0 +1,233 @@
+ * for structuring, mocking, and organizing test suites consistent with
+ * project conventions and accessibility standards.
+ */
+
+// Mock the SCSS module
+jest.mock("./page.module.scss", () => ({
+    page: "page",
+    header: "header",
+    title: "title",
+    add_button: "add_button",
+    pet_grid: "pet_grid",
+    pet_card: "pet_card",
+    pet_image: "pet_image",
+    image_icon: "image_icon",
+    pet_content: "pet_content",
+    pet_header: "pet_header",
+    pet_name: "pet_name",
+    edit_badge: "edit_badge",
+    info_row: "info_row",
+    info_label: "info_label",
+    info_value: "info_value",
+    view_details_button: "view_details_button",
+}));
+
+// Mock the pets data
+jest.mock("~data/pets/constants", () => ({
+    mockPets: [
+        {
+            id: "bella",
+            name: "Bella",
+            age: "5y 3m",
+            sex: "Female",
+            group: "Small animal",
+            species: "Dog",
+            breed: "Beagle",
+        },
+        {
+            id: "tweety",
+            name: "Tweety",
+            age: "7m",
+            sex: "Male",
+            group: "Bird",
+            species: "Cockatiel",
+            breed: "White-faced cockatiel",
+        },
+    ],
+}));
+
+describe("Pet Dashboard page", () => {
+    let user: ReturnType<typeof userEvent.setup>;
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        user = userEvent.setup();
+        render(<PetDashboard />);
+    });
+
+    it("renders", () => {
+        expect(
+            screen.getByRole("heading", { name: /my pets/i }),
+        ).toBeInTheDocument();
+    });
+
+    describe("Page header", () => {
+        it("displays the page title", () => {
+            const title = screen.getByRole("heading", { name: /my pets/i });
+            expect(title).toBeInTheDocument();
+        });
+
+        it("displays the add new pet button", () => {
+            const addButton = screen.getByRole("button", {
+                name: /add a new pet/i,
+            });
+            expect(addButton).toBeInTheDocument();
+        });
+
+        it("allows add button to be clicked", async () => {
+            const addButton = screen.getByRole("button", {
+                name: /add a new pet/i,
+            });
+            await user.click(addButton);
+            expect(addButton).toBeInTheDocument(); // Button should remain in document after click
+        });
+    });
+
+    describe("Pet cards display", () => {
+        it("renders all pet cards from mock data", () => {
+            // Check that both pet names appear
+            expect(screen.getByText("Bella")).toBeInTheDocument();
+            expect(screen.getByText("Tweety")).toBeInTheDocument();
+        });
+
+        it("displays correct number of pet cards", () => {
+            const petNames = screen.getAllByRole("heading", { level: 3 });
+            expect(petNames).toHaveLength(2);
+        });
+    });
+
+    describe("Pet information - Bella", () => {

```

Comment:
```suggestion
    describe("Pet information", () => {
```
Explained in the next comment, but I think we can make the description more generic here